pkg/credentialprovider/azure/azure_credentials.go

```diff
@@ -186,6 +191,30 @@ func (a *acrProvider) Enabled() bool {
 }
 
 func (a *acrProvider) Provide(image string) credentialprovider.DockerConfig {
+	a.mu.Lock()
+	defer a.mu.Unlock()
+
+	// If the cache hasn't expired, return our cache
+	if time.Now().Before(a.expiration) {
+		if a.config != nil && a.config.UseManagedIdentityExtension {
+			loginServer := a.parseACRLoginServerFromImage(image)
+			if loginServer == "" {
+				return a.cacheDockerConfig
+			}
+
+			// check whether loginServer is in cache
+			for k := range a.cacheDockerConfig {
+				if strings.EqualFold(loginServer, k) {
+					return a.cacheDockerConfig
+				}
+			}
+			// continue with get ARM token next since loginServer not found in cache
```

Comment:
Since we removed CachingDockerConfig, the code path to fetch ARM token would run for any non-ACR repo where the loginServer is never a match. Is that the case here?

Reply:
that's not correct, see Line 229: `klog.V(4).Infof("image(%s) is not from ACR, skip MSI authentication", image)`, if it's non-ACR repo, it won't fetch ARM token

Reply:
anyway, return empty DockerConfig for non-ACR repo looks reasonable, will fix it soon, thanks.

Reply:
fixed in line 202, PTAL.